pages/teams/[slug]/billing.tsx

```diff
@@ -0,0 +1,151 @@
+import { Error, Loading } from '@/components/shared';
+import { useTranslation } from 'next-i18next';
+import useTeam from 'hooks/useTeam';
+import { GetServerSidePropsContext } from 'next';
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
+import env from '@/lib/env';
+import { Button, Card } from 'react-daisyui';
+import useCanAccess from 'hooks/useCanAccess';
+import { TeamTab } from '@/components/team';
+import { useState } from 'react';
+import router from 'next/router';
+import fetcher from '@/lib/fetcher';
+import useSWR from 'swr';
+import ProductPricing from '@/components/payments/productPricing';
+import toast from 'react-hot-toast';
+
+const Payments = ({ teamFeatures }) => {
+  const { t } = useTranslation('common');
+  const { canAccess } = useCanAccess();
+  const { isLoading, isError, team } = useTeam();
+  const { data } = useSWR(
+    team?.slug ? `/api/teams/${team?.slug}/payments/products` : null,
+    fetcher
+  );
+  const [portalLinkLoading, setPortalLinkLoading] = useState(false);
+
+  const postData = async ({
+    url,
+    data,
+  }: {
+    url: string;
+    data?: { price: any };
+  }) => {
+    setPortalLinkLoading(true);
+    const res = await fetch(url, {
+      method: 'POST',
+      headers: new Headers({ 'Content-Type': 'application/json' }),
+      credentials: 'same-origin',
+      body: JSON.stringify(data),
+    });
+
+    if (!res.ok) {
+      toast.error(`${t('error-occurred')}`);
+    }
+    return res.json();
+  };
+
+  const redirectToCustomerPortal = async () => {
+    try {
+      const { data: url } = await postData({
+        url: `/api/teams/${team?.slug}/payments/create-portal-link`,
+      });
+      router.push(url);
+    } catch (error) {
+      setPortalLinkLoading(false);
+      if (error) return alert((error as Error).message);
+    }
+  };
+
+  if (isLoading) {
+    return <Loading />;
+  }
+
+  if (isError) {
+    return <Error message={isError.message} />;
+  }
+
+  if (!team) {
+    return <Error message={t('team-not-found')} />;
+  }
+  return (
+    <>
+      {canAccess('team_payments', ['read']) && (
+        <>
+          <TeamTab
+            activeTab="payments"
+            team={team}
+            teamFeatures={teamFeatures}
+          />
+          <div className="">
+            <Button
+              type="button"
+              color="primary"
+              size="md"
+              loading={portalLinkLoading}
+              onClick={redirectToCustomerPortal}
+            >
+              {t('open-customer-portal')}
+            </Button>
```

Comment:
Should we hide the link to customer portal if there is no subscription yet or customer id created?

Reply:
When user clicks on the button to open customer portal, in the api call we check if customer is registered on stripe or not, if not we create a customer on stripe save the customerId and open portal